### PR TITLE
Feature/add category entity and endpoint

### PR DESCRIPTION
**Pull request by ticklemynausea**
This pull request does the following changes

- Moves the indicator categories/subcategories into its own Entity (Category)
- Updates the user interface to accomodate these changes
- Adds `/api/v1/categories` and a `category` query param filter to `/api/v1/indicators` that searches by category/subcategory id

### CODE REVIEW

File: app/models/category.rb [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+      category = category.first
+
+      unless category
+        category = Category.new(attributes)
+      end
+
+      category
+    end
+
+    def fetch_all(options)
```
Comment (qoobaa) [inline app/models/category.rb]: I tried to move the logic behind filtering models into separate services, and I'd try to do it here as well. It'd allow reducing the size of the `Category` model significantly.

File: app/controllers/api/v1/categories_controller.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+module Api
+  module V1
+    class CategoriesController < ApiController
+      def index
+        categories = Category.
```
Comment (qoobaa) [inline app/controllers/api/v1/categories_controller.rb]: Is it really necessary to make it 3 lines here?

Comment (ticklemynausea) [reply app/controllers/api/v1/categories_controller.rb]: No, I can change that.

File: app/models/category.rb (hunk 1 above)
Comment (ticklemynausea) [reply app/models/category.rb]: Sounds good to me. Should we do this after merging?

Comment (qoobaa) [reply app/models/category.rb]: Sure, we can do it in a separate PR. Have a look here: https://github.com/Vizzuality/emissions-scenario-portal/pull/161 to get the idea.

**Pull request merged**